doc/cloud-init.md

```diff
@@ -92,6 +92,35 @@ config:
 See {ref}`How to validate user data <cloud-init:check_user_data_cloud_config>` for information on how to check whether the syntax is correct.
 ```
 
+### Configure `cloud-init` through the API
+
+If you are using the API to configure your instance, provide the `cloud-init` configuration as a string with escaped newline characters.
+
+For example:
+
+    lxc query --request PATCH /1.0/instances/<instance_name> --data '{
+      "config": {
+        "cloud-init.user-data": "#cloud-config\npackage_upgrade: true\npackages:\n  - package1\n  - package2"
+      }
+    }'
+
+Alternatively, to avoid mistakes, write the configuration to a file and include that in your request.
+For example, create `cloud-init.txt` with the following content:
+
+    #cloud-config
+    package_upgrade: true
+    packages:
+      - package1
+      - package2
+
+Then send the following request:
+
+    lxc query --request PATCH /1.0/instances/<instance_name> --data '{
+    "config": {
+      "cloud-init.user-data": "'"$(awk -v ORS='\\n' '1' cloud-init.txt)"'"
```

Comment:
What does this awk line do btw?

Reply:
It replaces the line breaks with `\n` and thus converts the file into a one-line string.

Reply:
Had to get some help on this one, since I got the line from Google ... seems to make sense. ;)

![image](https://github.com/canonical/lxd/assets/11227796/b934665b-6c41-4204-8bd1-1d1583f17551)